CONTRIBUTING.md

```diff
@@ -76,37 +76,66 @@ Check out our dedicated [Bevy Organization document](/docs/the_bevy_organization
-* Typo and grammar fixes.
+PRs with non trivial design decisions are given the [`S-Controversial`] label. This indicates that
+the PR should either go through an RFC (although in some cases small enough decisions can get by without).
 
-PRs are controversial if there is serious design discussion required, or a large impact to contributors or users. Factors that increase controversy include:
+PRs that are non trivial to review are given the [`D-Complex`] label. This indicates that the PR
+should be reviewed more thoroughly and by people with experience in the area that the PR touches.
+
+When making PRs, try to split out more controversial changes from less controversial ones, in order to make your work easier to review and merge.
+It is also a good idea to try and split out simple changes from more complex changes if it is not helpful for then to be reviewed together.
 
-1. Changes to project-wide workflow or style.
+Some things that are reason to apply the [`S-Controversial`] label to a PR:
+
+1. Changes to a project-wide workflow or style.
 2. New architecture for a large feature.
-3. PRs where a serious tradeoff must be made.
+3. Serious tradeoffs were made.
 4. Heavy user impact.
 5. New ways for users to make mistakes (footguns).
-6. Introductions of `unsafe` code.
-7. Large-scale code reorganization.
-8. High levels of technical complexity.
-9. Adding a dependency.
-10. Touching licensing information (due to the level of precision required).
-11. Adding root-level files (due to the high level of visibility).
+6. Adding a dependency
+7. Touching licensing information (due to level of precision required).
+8. Adding root-level files (due to the high level of visibility)
+
+Some things that are reason to apply the [`D-Complex`] label to a PR:
+
+1. Introduction or modification of soundness relevent code (for example `unsafe` code)
+2. High levels of technical complexity.
+3. Large-scale code reorganization
+
+Examples of PRs that are not [`S-Controversial`] or [`D-Complex`]:
+
+* Fixing dead links.
+* Removing dead code or unused dependencies.
+* Typo and grammar fixes.
+* [Add `Mut::reborrow`](https://github.com/bevyengine/bevy/pull/7114)
+* [Add `Res::clone`](https://github.com/bevyengine/bevy/pull/4109)
+
+Examples of PRs that are [`S-Controversial`] but not [`D-Complex`]:
+
+* [Implement and require `#[derive(Component)]` on all component structs](https://github.com/bevyengine/bevy/pull/2254)
+* [Use default serde impls for Entity](https://github.com/bevyengine/bevy/pull/6194)
+
+Examples of PRs that are not [`S-Controversial`] but are [`D-Complex`]:
+
+* [Ensure `Ptr`/`PtrMut`/`OwningPtr` are aligned in debug builds](https://github.com/bevyengine/bevy/pull/7117)
+* [Replace `BlobVec`'s `swap_scratch` with a `swap_nonoverlapping`](https://github.com/bevyengine/bevy/pull/4853)
+
+Examples of PRs that are both [`S-Controversial`] and [`D-Complex`]:
+
+* [bevy_reflect: Binary formats](https://github.com/bevyengine/bevy/pull/6140)
```

Comment:
I really like this expansion with concrete example PRs. Helps ground the entire section.